src/components/MultiHopTrade/components/Approval/Approval.tsx

```diff
@@ -0,0 +1,285 @@
+                      color='blue.500'
+                      href={`${tradeQuoteStep.sellAsset.explorerTxLink}${approvalTxId}`}
+                    >
+                      <MiddleEllipsis value={approvalTxId} />
+                    </Link>
+                  </Row.Value>
+                </Row>
+              )}
+              <Row>
+                <Row.Label display='flex' alignItems='center'>
+                  <Text color='gray.500' translation='trade.allowance' />
+                  <Tooltip label={translate('trade.allowanceTooltip')}>
+                    <Box ml={1}>
+                      <Icon as={FaInfoCircle} color='gray.500' fontSize='0.7em' />
+                    </Box>
+                  </Tooltip>
+                </Row.Label>
+                <Row.Value textAlign='right' display='flex' alignItems='center'>
+                  <Text
+                    color={isExactAllowance ? 'gray.500' : 'white'}
+                    translation='trade.unlimited'
+                    fontWeight='bold'
+                  />
+                  <Switch
+                    size='sm'
+                    mx={2}
+                    isChecked={isExactAllowance}
+                    onChange={toggleIsExactAllowance}
+                  />
+                  <Text
+                    color={isExactAllowance ? 'white' : 'gray.500'}
+                    translation='trade.exact'
+                    fontWeight='bold'
+                  />
+                </Row.Value>
+              </Row>
+              <Button type='submit' size='lg' isLoading={isLoading} colorScheme='blue' mt={2}>
+                <Text translation='common.confirm' />
+              </Button>
+              {!approvalTxId && !isSubmitting && (
+                <Button variant='ghost' mt={2} size='lg' onClick={() => history.goBack()}>
+                  <Text translation='common.reject' />
+                </Button>
+              )}
+              <Divider my={4} />
+              <Row>
+                <Row.Label>
+                  <Text color='gray.500' translation='trade.estimatedGasFee' />
+                </Row.Label>
+                <Row.Value textAlign='right'>
+                  <Skeleton isLoaded={approvalNetworkFeeCryptoHuman !== undefined}>
+                    <RawText>{approvalNetworkFeeFiatDisplay}</RawText>
+                    <RawText color='gray.500'>{approvalNetworkFeeCryptoHumanDisplay}</RawText>
+                  </Skeleton>
+                </Row.Value>
+              </Row>
+            </Flex>
+          </Flex>
+        </Card.Body>
+      </Card>
+    </SlideTransition>
+  )
+}
+
+export const Approval = () => {
+  const history = useHistory()
+  const tradeQuoteStep = useAppSelector(selectFirstHop)
+  const swapperName = useAppSelector(selectSelectedSwapperName)
+
+  if (!tradeQuoteStep || !swapperName) {
+    history.goBack()
+    return null
+  }
+
+  return <ApprovalInner tradeQuoteStep={tradeQuoteStep} swapperName={swapperName} />
+}
```

Comment:
this wrapper allows effective conditional call of hooks. We want to exit if the tradequotestep is undefined to simplify logic in the actual approval component (otherwise we end up with heaps if optional chaining etc)